bench.zig

```diff
@@ -156,9 +156,13 @@ pub fn benchmarkArgs(comptime name: []const u8, comptime f: anytype, comptime ar
     inline for (args) |a| {
         var ctx = Context.init();
 
+        comptime const arg_type = argTypeFromFn(f);
+        comptime const bench_fn_type: type = fn (*Context, arg_type) callconv(.Async) void;
+        comptime const f2: bench_fn_type = f;
+
         var result: void = undefined;
-        var frame: @Frame(f) = undefined;
-        _ = @asyncCall(&frame, &result, f, .{&ctx, a});
+        var frame: @Frame(f2) = undefined;
+        _ = @asyncCall(&frame, &result, f2, .{&ctx, a});
```

Comment:
Technically, this casts any non-async function to an async one. It doesn't make a difference if you use an async benchmark or not.

Reply:
That's what I figured, but is it guaranteed that the call will not be inlined ?
Maybe `@asyncCall` effectively does this but it's not obvious from the docs and I haven't played with it so no experience on my side.

It's important that the benchmarked function call is not inlined base on the parameters as this would modify the results.

Reply:
I'm pretty sure that async functions can't be inlined. Internally, they use an enum for the Callconv. It has the callconv for inline, C, async, undefined, and a bunch of others. What we are doing here is trying to cast undefined to async, I think.

Reply:
It'd be great if you could demonstrate it with a simple example from https://godbolt.org.